### PR TITLE
This makes the ICU requirement less hostile.

### PR DESCRIPTION
**Pull request by lemire**
This will make building ada a bit nicer with respect to the ICU dependency.

(It could be improved further.)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -38,7 +38,38 @@ set_target_properties(
...
+endif()
+
+## If we put 'REQUIRED' under find_package, it is a hard error and
+## we get no chance to provide instructions to the user.
+find_package(ICU COMPONENTS uc i18n)
+
+### If the user does not have ICU, let us help them with instructions:
+if (NOT ICU_FOUND)
+    if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+      if(EXISTS /opt/homebrew)
```
Comment (anonrig) [inline CMakeLists.txt:60]: Intel based macOS has a different path for homebrew.

Comment (lemire) [reply CMakeLists.txt:64]: Fixed.

**Comment by lemire** [comment]
I'll merge this because I do not think it can cause much harm and it should improve quality of life.

**Comment by anonrig** [comment]
This is definitely a good improvement. Thanks.

**Pull request merged**